functions/go/gatekeeper-validate/go.mod

```diff
@@ -0,0 +1,11 @@
+module github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/gatekeeper-validate
+
+go 1.15
+
+require (
+	github.com/open-policy-agent/frameworks/constraint v0.0.0-20210317225149-4f80ac172ddf
+	github.com/open-policy-agent/gatekeeper v3.0.4-beta.2+incompatible
```

Comment:
We should use the most recent version of Gatekeeper that makes sense.

v3.0.4 is probably missing a few matchers, such as the scope matcher.

https://github.com/open-policy-agent/gatekeeper/releases/tag/v3.3.0

is probably a good starting point.

Reply:
I got the following error when trying to get it directly.
```
$ go get github.com/open-policy-agent/gatekeeper@v3.3.0
go get github.com/open-policy-agent/gatekeeper@v3.3.0: github.com/open-policy-agent/gatekeeper@v3.3.0: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v3
```

It seems it is because it has a semver major version of 2 or higher and is not a Go module yet.
I then got it using the hash ofv3.3.0 directly. It works, but it doesn't look so pretty in go.mod.

Reply:
Yeah, unfortunately there was a decision early on that it should be "v3" because there were 2 previous ways to integrate OPA with Kubernetes. I wish I had fought harder against that.

Gomod requires a specific directory structure for major versions >2, where the code must live in a subfolder `/vN` where `N` is the major version number. You'll need to use the hash.

At some point, I'd like to take the `target` code out of Gatekeeper and put it in its own repo, which should fix this.